subprojects/shipkit/src/main/groovy/org/shipkit/internal/notes/util/IOUtil.java

```diff
@@ -80,31 +80,26 @@ public static void createDirectory(File file) {
      */
     public static void downloadToFile(String url, File file) {
         InputStream input = null;
-        FileOutputStream output = null;
         try {
             input = new BufferedInputStream(new URL(url).openStream());
 
             IOUtil.createParentDirectory(file);
 
-            FileOutputStream fos = new FileOutputStream(file);
-            byte[] buf = new byte[1024];
-            int n;
-            while ((n = input.read(buf)) != -1) {
-                fos.write(buf, 0, n);
+            FileOutputStream fos = null;
+            try {
+                fos = new FileOutputStream(file);
+                byte[] buf = new byte[1024];
+                int n;
+                while ((n = input.read(buf)) != -1) {
+                    fos.write(buf, 0, n);
+                }
+            } finally {
+                close(fos);
```

Comment:
If close() fails with IOException, we are losing the original exception.

Perhaps we need 2 methods: "close()" and "closeQuietly()".

I know that this problem was present before... since you are changing the code now, let's improve it :)

Reply:
+1